### PR TITLE
Check auditd is running and enabled

### PR DESCRIPTION
**Pull request by vijaynadella**

### CODE REVIEW

File: CollectionMonitor.cpp [hunk 1]
```diff
@@ -192,7 +193,10 @@ void CollectionMonitor::signal_collector(int sig) {
 }
 
 bool CollectionMonitor::is_auditd_present() {
-    return PathExists(_auditd_path);
+    int auditd_present = std::system("which auditd > /dev/null 2>&1");
+    int auditd_enabled = std::system("systemctl is-enabled auditd.service > /dev/null 2>&1");
+    int auditd_active = std::system("systemctl is-active auditd.service > /dev/null 2>&1");
```
Comment (bmanikm) [inline CollectionMonitor.cpp:198]: This assumes that the init system is systemd. What about upstart and sysvinit?

[the remark this answers is not in the document]
Comment (vijaynadella) [reply CollectionMonitor.cpp:198]: Returns 0 if the service is enabled or active, else a random integer.

File: build/docker/auoms-build32/Dockerfile [hunk 2]
```diff
@@ -32,4 +32,4 @@ RUN yum install -y util-linux-ng.i686 \
 RUN sed -i '/requiretty/d' /etc/sudoers \
     && echo "build ALL=(ALL) NOPASSWD:ALL" >>/etc/sudoers
 
-ENTRYPOINT ["/usr/bin/linux32"]
\ No newline at end of file
+ENTRYPOINT ["/usr/bin/linux32"]
```
Comment (bmanikm) [inline build/docker/auoms-build32/Dockerfile:35]: Consider reverting the new line change in this file

File: CollectionMonitor.cpp [hunk 3]
```diff
@@ -191,8 +193,44 @@ void CollectionMonitor::signal_collector(int sig) {
...
+    if (!file.is_open()) {
+        return false;
+    }
+
+    std::string line;
+    while (std::getline(file, line)) {
+        // Check if the line contains 'start on' indicating service is enabled
+        if (line.find("start on") != std::string::npos) {
+            file.close();
+            isEnabledStatus = 1;
```
Comment (bmanikm) [inline CollectionMonitor.cpp:220]: Consider break after this

File: CollectionMonitor.cpp [hunk 4]
```diff
@@ -191,8 +193,44 @@ void CollectionMonitor::signal_collector(int sig) {
...
+    std::ifstream file("/etc/init/auditd.conf");
+    if (!file.is_open()) {
+        return false;
+    }
+
+    std::string line;
+    while (std::getline(file, line)) {
+        // Check if the line contains 'start on' indicating service is enabled
+        if (line.find("start on") != std::string::npos) {
+            file.close();
```
Comment (bmanikm) [inline CollectionMonitor.cpp:219]: Do we need this given file.close() at 223 line?

**Review by bmanikm — APPROVED** [review]
:shipit:

**Pull request merged**